Source/VMware.vSphereDSC/DSCResources/PowerCLISettings.ps1

```diff
@@ -0,0 +1,235 @@
+
+        $this.PopulateResult($powerCLICurrentConfiguration, $result)
+
+        return $result
+    }
+
+    <#
+    .DESCRIPTION
+
+    Imports the needed VMware Modules.
+    #>
+    [void] ImportRequiredModules() {
+        $savedVerbosePreference = $global:VerbosePreference
+        $global:VerbosePreference = 'SilentlyContinue'
+
+        Import-Module -Name VMware.VimAutomation.Core
+
+        $global:VerbosePreference = $savedVerbosePreference
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns all passed PowerCLI configuration properties as a hashtable.
+    #>
+    [hashtable] GetPowerCLIConfigurationProperties() {
+        $powerCLIConfigurationProperties = @{}
+
+        # Adds the Default Scope to the hashtable.
+        $powerCLIConfigurationProperties.Add("Scope", $this.Scope)
+
+        if ($this.CEIPDataTransferProxyPolicy -ne [ProxyPolicy]::Unset -and $this.ParticipateInCeip -eq $true) {
+            $powerCLIConfigurationProperties.Add("CEIPDataTransferProxyPolicy", $this.CEIPDataTransferProxyPolicy)
+        }
+
+        if ($this.DefaultVIServerMode -ne [DefaultVIServerMode]::Unset) {
+            $powerCLIConfigurationProperties.Add("DefaultVIServerMode", $this.DefaultVIServerMode)
+        }
+
+        if ($null -ne $this.DisplayDeprecationWarnings) {
+            $powerCLIConfigurationProperties.Add("DisplayDeprecationWarnings", $this.DisplayDeprecationWarnings)
+        }
+
+        if ($this.InvalidCertificateAction -ne [BadCertificateAction]::Unset) {
+            $powerCLIConfigurationProperties.Add("InvalidCertificateAction", $this.InvalidCertificateAction)
+        }
+
+        if ($null -ne $this.ParticipateInCeip) {
+            $powerCLIConfigurationProperties.Add("ParticipateInCeip", $this.ParticipateInCeip)
+        }
+
+        if ($this.ProxyPolicy -ne [ProxyPolicy]::Unset) {
+            $powerCLIConfigurationProperties.Add("ProxyPolicy", $this.ProxyPolicy)
+        }
+
+        if ($null -ne $this.WebOperationTimeoutSeconds) {
+            $powerCLIConfigurationProperties.Add("WebOperationTimeoutSeconds", $this.WebOperationTimeoutSeconds)
+        }
+
+        return $powerCLIConfigurationProperties
+    }
+
+    <#
+    .DESCRIPTION
+
+    Constructs the Set-PowerCLIConfiguration cmdlet with the passed properties.
+    #>
+    [string] ConstructCommandWithParameters($commandName, $properties, $namesOfProperties) {
+        $constructedCommand = ''
+
+        # Adds the command name to the constructed command.
+        $constructedCommand += "$commandName "
+
+        # For every property name we add the property value with the following syntax: '-Property Value'.
+        foreach ($propertyName in $namesOfProperties) {
+            $propertyValue = $properties.$propertyName
+
+            # For bool values we need to add another '$' sign so the value can be evaluated to bool.
+            if ($propertyValue.GetType().Name -eq 'Boolean') {
```

Comment:
This hack will work until the moment we introduce setting of complex type which could be never, but at least document the hack